### PR TITLE
Support version 1.20

### PR DESCRIPTION
**Pull request by phracek**

### CODE REVIEW

**Comment by phracek** [comment]
Let's try also Testing Farm

[test]

**Comment by phracek** [comment]
[test]

**Comment by phracek** [comment]
Fedora tests are failing for this reason:

```bash
Removing intermediate container 165ae2d1b794
Step 3/7 : ADD nginx-container/examples/1.20/test-app/nginx.conf "${NGINX_CONF_PATH}"
lstat nginx-container/examples/1.20/test-app/nginx.conf: no such file or directory
ERROR: The image cannot be built from /tmp/tmp.GLwLrkloMH and application https://github.com/sclorg/nginx-container.git.
Terminating the Dockerfile build.
```

The test expects that new version already exists in the `examples` directory, which is not true https://github.com/sclorg/nginx-container/blob/master/test/run#L421
The clone is done here: https://github.com/sclorg/container-common-scripts/blob/master/test-lib.sh#L879

**Comment by phracek** [comment]
[test]

**Comment by phracek** [comment]
[test-openshift][test-openshift-4]

**Comment by phracek** [comment]
Tests on OpenShift 3 are failing for this reason:

```bash
command POD not running yet, will start one called command-app ubi7/ubi
pod/command-app created
Waiting for command POD .NAME                                 READY     STATUS              RESTARTS   AGE
command-app                          0/1       ContainerCreating   0          1s
nginx-116-centos7-testing-1-build    0/1       Completed           0          21s
nginx-116-centos7-testing-1-n2w7g    1/1       Terminating         0          11s
nginx-116-centos7-testing-2-build    0/1       Completed           0          12s
nginx-116-centos7-testing-2-deploy   1/1       Running             0          5s
nginx-116-centos7-testing-2-rnvs6    1/1       Running             0          3s
.NAME                                READY     STATUS         RESTARTS   AGE
command-app                         0/1       ErrImagePull   0          4s
nginx-116-centos7-testing-1-build   0/1       Completed      0          24s
nginx-116-centos7-testing-1-n2w7g   0/1       Terminating    0          14s
nginx-116-centos7-testing-2-build   0/1       Completed      0          15s
nginx-116-centos7-testing-2-rnvs6   1/1       Running        0          6s
.NAME                                READY     STATUS         RESTARTS   AGE
command-app                         0/1       ErrImagePull   0          8s
nginx-116-centos7-testing-1-build   0/1       Completed      0          28s
nginx-116-centos7-testing-1-n2w7g   0/1       Terminating    0          18s
nginx-116-centos7-testing-2-build   0/1       Completed      0          19s
nginx-116-centos7-testing-2-rnvs6   1/1       Running        0          10s
.NAME                                READY     STATUS             RESTARTS   AGE
command-app                         0/1       ImagePullBackOff   0          28s
nginx-116-centos7-testing-1-build   0/1       Completed          0          48s
nginx-116-centos7-testing-2-build   0/1       Completed          0          39s
nginx-116-centos7-testing-2-rnvs6   1/1       Running            0          30s
```

File: 1.20/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    yum -y clean all --enablerepo='*'
+
+# Copy the S2I scripts from the specific language image to $STI_SCRIPTS_PATH
+COPY ./s2i/bin/ $STI_SCRIPTS_PATH
+
+# Copy extra files to the image.
+COPY ./root/ /
+
+# In order to drop the root user, we have to make some directories world
+# writeable as OpenShift default security model is to run the container under
```
Comment (ljavorsk) [inline 1.20/Dockerfile:68]: TYPO, writable

Comment (ljavorsk) [reply 1.20/Dockerfile:68]: There are 4 occurrences of this TYPO

File: 1.20/README.md [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+a new image that runs the assembled application as an output.
+
+In case of Nginx container image, the application source code is typically
+either static HTML pages or configuration files.
+
+To support the Source-to-Image framework, important scripts are included in the builder image:
+
+* The `/usr/libexec/s2i/run` script is set as the default command in the resulting container image (the new image with the application artifacts).
+
+* The `/usr/libexec/s2i/assemble` script inside the image is run to produce a new image with the application artifacts. The script takes sources of a given application (HTML pages), Nginx configuration files, and places them into appropriate directories inside the image. The structure of nginx-app can look like this:
```
Comment (ljavorsk) [inline 1.20/README.md:54]: TYPO, is ran (Passive voice)

File: 1.20/README.md [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+
+#### 3. Prepare an application inside a container
+
+This step usually consists of at least these parts:
+
+* putting the application source into the container
+* moving configuration files to the correct place (if available in the application source code)
+* setting the default command in the resulting image
+
+For all these three parts, you can either set up all manually and use the `nginx` command explicitly in the Dockerfile ([3.1.](#31-to-use-own-setup-create-a-dockerfile-with-this-content)), or you can use the Source-to-Image scripts inside the image ([3.2.](#32-to-use-the-source-to-image-scripts-and-build-an-image-using-a-dockerfile-create-a-dockerfile-with-this-content). For more information about these scripts, which enable you to set-up and run the nginx daemon, see the "Source-to-Image framework and scripts" section above.
```
Comment (ljavorsk) [inline 1.20/README.md:105]: As reading this sentence "For more information..." I'm not able to tell what you mean by the part "which enable you to". Consider rephrasing it.

File: 1.20/README.md [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+# Run script uses standard ways to run the application
+CMD nginx -g "daemon off;"
+```
+
+##### 3.2. To use the Source-to-Image scripts and build an image using a Dockerfile, create a Dockerfile with this content:
+
+```
+FROM registry.access.redhat.com/ubi8/nginx-120
+
+# Add application sources to a directory that the assemble script expects them
```
Comment (ljavorsk) [inline 1.20/README.md:127]: Change the word 'that' for 'where'

File: 1.20/README.md [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+**If `NGINX_LOG_TO_VOLUME` variable is set, nginx logs into `/var/log/nginx/`. In case of RHEL-7 and CentOS-7 images, this is a symlink to `/var/opt/rh/rh-nginx120/log/nginx/`, which can be mounted to host system using the container volumes.**
+
+
+See also
+--------
+Dockerfile and other sources for this container image are available on
+https://github.com/sclorg/nginx-container.
+In that repository you also can find another versions of Python environment Dockerfiles.
+Dockerfile for CentOS is called `Dockerfile`, Dockerfile for RHEL7 is called `Dockerfile.rhel7`,
+for RHEL8 it's `Dockerfile.rhel8` and the Fedora Dockerfile is called Dockerfile.fedora.
```
Comment (ljavorsk) [inline 1.20/README.md:206]: Be consistent and wrap 'Dockerfile.fedora' into quotes.

File: 1.20/README.md (hunk 3 above)
Comment (phracek) [reply 1.20/README.md]: Rephrased

**Comment by phracek** [comment]
[test][test-openshift]

**Comment by phracek** [comment]
Because of Infra problems restarting tests.

[test][test-openshift]

**Comment by phracek** [comment]
CentOS7 OpenShift 3 test is failing because of problem with pulling ubi7/ubi image from RedHat registry. This issue should be fixed by https://github.com/sclorg/container-common-scripts/pull/198

File: 1.20/README.md [hunk 6]
```diff
@@ -51,7 +51,7 @@ To support the Source-to-Image framework, important scripts are included in the
 
 * The `/usr/libexec/s2i/run` script is set as the default command in the resulting container image (the new image with the application artifacts).
 
-* The `/usr/libexec/s2i/assemble` script inside the image is run to produce a new image with the application artifacts. The script takes sources of a given application (HTML pages), Nginx configuration files, and places them into appropriate directories inside the image. The structure of nginx-app can look like this:
+* The `/usr/libexec/s2i/assemble` script inside the image is ran to produce a new image with the application artifacts. The script takes sources of a given application (HTML pages), Nginx configuration files, and places them into appropriate directories inside the image. The structure of nginx-app can look like this:
```
Comment (pkubatrh) [inline 1.20/README.md:54]: The original is correct

File: 1.20/README.md [hunk 7]
```diff
@@ -102,7 +102,7 @@ This step usually consists of at least these parts:
 * moving configuration files to the correct place (if available in the application source code)
 * setting the default command in the resulting image
 
-For all these three parts, you can either set up all manually and use the `nginx` command explicitly in the Dockerfile ([3.1.](#31-to-use-own-setup-create-a-dockerfile-with-this-content)), or you can use the Source-to-Image scripts inside the image ([3.2.](#32-to-use-the-source-to-image-scripts-and-build-an-image-using-a-dockerfile-create-a-dockerfile-with-this-content). For more information about these scripts, which enable you to set-up and run the nginx daemon, see the "Source-to-Image framework and scripts" section above.
+For all these three parts, you can either set up all manually and use the `nginx` command explicitly in the Dockerfile ([3.1.](#31-to-use-own-setup-create-a-dockerfile-with-this-content)), or you can use the Source-to-Image scripts inside the image ([3.2.](#32-to-use-the-source-to-image-scripts-and-build-an-image-using-a-dockerfile-create-a-dockerfile-with-this-content), that already know how to set-up and run some common Nginx applications.
```
Comment (pkubatrh) [inline 1.20/README.md:105]: Why was the reference to the "Source-to-Image framework and scripts" section removed?

**Comment by pkubatrh** [comment]
Some comments regarding the doc changes, otherwise LGTM

File: 1.20/README.md (hunk 7 above)
Comment (phracek) [reply 1.20/README.md:105]: The same documentation style like here: https://github.com/sclorg/s2i-perl-container/tree/master/5.30#3-prepare-an-application-inside-a-container or here: https://github.com/sclorg/s2i-nodejs-container/tree/master/14#3-prepare-an-application-inside-a-container . But some part is still missing. I will fix it like in `s2i-perl-container` .

**Comment by phracek** [comment]
[test]

File: 1.20/README.md (hunk 6 above)
Comment (pkubatrh) [reply 1.20/README.md]: This is still unresolved
```suggestion
* The `/usr/libexec/s2i/assemble` script inside the image is run to produce a new image with the application artifacts. The script takes sources of a given application (HTML pages), Nginx configuration files, and places them into appropriate directories inside the image. The structure of nginx-app can look like this:
```

Comment (pkubatrh) [reply 1.20/README.md]: Ignore please, somehow I managed to look at old state even after refresh

**Comment by pkubatrh** [comment]
lgtm now. Are we testing the new version in the CI, or only the changes for released versions?

**Comment by phracek** [comment]
> testing it

Definitely yes, we can add there a file `.devel-repo-rhel7` and delete `.exclude-rhel7` file. I will add it and restart the tests.

**Comment by phracek** [comment]
> lgtm now. Are we testing the new version in the CI, or only the changes for released versions?
> 
> Edit: I see .exclude-rhel7 so I guess not. Can we enable testing it then with pre-release content?

Definitely yes, we can add there a file `.devel-repo-rhel7` and delete the `.exclude-rhel7` file. I will add it and restart the tests.

**Comment by phracek** [comment]
[test]

**Comment by phracek** [comment]
Devel repo has been updated. Let's try one more time.

[test]

**Comment by phracek** [comment]
The tests are GREEN. Let's get merge it.

**Pull request merged**